### ISSUE TITLE
Exception type and message in test for physicalField.itemset

### ISSUE THREAD

**Issue by shwina** [usnistgov/fipy#568]
In https://github.com/usnistgov/fipy/blob/develop/fipy/tools/dimensions/physicalField.py#L601, should the exception be `ValueError`?

See below, `a.itemset(b)` raises a `ValueError`, while `a.itemset(b.value)` raises `IndexError`:

```
In [2]: from fipy import *

In [4]: a = PhysicalField(((4, 5), (6, 7)), 'm')

In [5]: a
Out[5]:
PhysicalField(array([[4, 5],
       [6, 7]]),'m')

In [6]: a.itemset(PhysicalField("6 ft"))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-8292a0773678> in <module>()
----> 1 a.itemset(PhysicalField("6 ft"))

/home/atrikut/local/fipy/fipy/tools/dimensions/physicalField.pyc in itemset(self, value)
    614         if isinstance(value, PhysicalField) or _isVariable(value):
    615             value = self._inMyUnits(value).value
--> 616         self.value.itemset(value)
    617
    618

ValueError: can only convert an array of size 1 to a Python scalar

In [7]: a.itemset(PhysicalField("6 ft").value)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-7-d21047dad36c> in <module>()
----> 1 a.itemset(PhysicalField("6 ft").value)

/home/atrikut/local/fipy/fipy/tools/dimensions/physicalField.pyc in itemset(self, value)
    614         if isinstance(value, PhysicalField) or _isVariable(value):
    615             value = self._inMyUnits(value).value
--> 616         self.value.itemset(value)
    617
    618

IndexError: index 47828697664376 is out of bounds for axis 0 with size 2

In [8]:
```

**Comment by guyer** [comment]
As the comment says in the code you linked, different versions of NumPy raise different exceptions. The test is set up to try to normalize this fact.

**Comment by shwina** [comment]
Thanks, that makes sense.

With latest NumPy (1.14), the test currently fails with the following output:

```
Failed example:
    try:
        a.itemset(PhysicalField("6 ft"))
    except IndexError:
        # NumPy 1.7 has changed the exception type
        raise ValueError("can only place a scalar for an  array of size 1")
Expected:
    Traceback (most recent call last):
        ...
    ValueError: can only place a scalar for an  array of size 1
Got:
    Traceback (most recent call last):
      File "/software/anaconda/5.1.0/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest fipy.tools.dimensions.physicalField.PhysicalField.itemset[4]>", line 2, in <module>
        a.itemset(PhysicalField("6 ft"))
      File "/home/atrikut/local/fipy/fipy/tools/dimensions/physicalField.py", line 616, in itemset
        self.value.itemset(value)
    ValueError: can only convert an array of size 1 to a Python scalar
```

**Comment by guyer** [comment]
Fixed by #607
